packages/x-adapter/README.md

```diff
@@ -1,13 +1,218 @@
+request/response `mappers`.
+
+Take in account that if your use our `SearchRequest` and `SearchResponse`
+[search-types](https://github.com/empathyco/x/tree/main/packages/search-types), you will need to map
+some fields as they are required in the model.
+
+```tsx
+import { SearchRequest, SearchResponse } from '@empathyco/x-types';
+import { EndpointAdapterOptions } from '@empathyco/x-adapter';
+
+export const searchEndpointAdapter: EndpointAdapterOptions<SearchRequest, SearchResponse> = {
+  endpoint: `https://${API_ENDPOINT}`,
+  defaultRequestOptions: {
+    properties: {
+      headers: { Authorization: `${YOUR_AUTH}` }
+      // ... any options your API may need
+    }
+  },
+  requestMapper({ query, start, rows }) {
+    return {
+      q: query,
+      limit: rows,
+      offset: start
+      // ... any field you need to send for your request
+    };
+  },
+  responseMapper({ results }) {
+    return {
+      results: results,
+      totalResults: results.length
+      // ... any field you need to map from the response
+    };
+  }
+};
+```
+
+#### Using a factory function to configure your adapter
+
+To ease this configuration, you can also use the `x-adapter` ’s `endpointAdapterFactory` function
+that defines how a request will be triggered and resolved with the given `EndpointAdapterOptions`.
+
+```tsx
+import { endpointAdapterFactory } from '@empathyco/x-adapter';
+import { SearchRequest, SearchResponse } from '@empathyco/x-types';
+
+export const searchEndpointAdapter = endpointAdapterFactory<SearchRequest, SearchResponse>({
+  endpoint: 'https://{API_ENDPOINT}',
+  defaultRequestOptions: {
+    properties: {
+      headers: { Authorization: `${YOUR_AUTH}` }
+      // ... any options your API may need
+    }
+  },
+  // we enhance you to declare your resquest/response mappers ouside this function to improve its legibility
+  requestMapper: searchRequestMapper,
+  responseMapper: searchResponseMapper
+});
+```
+
+### Implement your own adapter using schemas
+
+The `x-adapter` library includes a `schemaMapperFactory` function that creates a mapper function to
+execute the schema you provide. You just need to create your request and response schemas to define
+the structure of your data, and pass them as a parameter to the factory function.
+
+```tsx
+import { schemaMapperFactory } from '@empathyco/x-adapter';
+import { SearchRequest } from '@empathyco/x-types';
+import { searchRequestSchema } from '../../schemas/requests/search-request.schema';
+
+// Import and use already existing types from your API source, or declare your own based on the response you get.
+interface ApiRequestModel {
+  q: string;
+  limit?: number;
+  offset?: number;
+}
+interface ApiResponseModel {
+  items: T[];
+  limit?: number;
+  offset?: number;
+  total: number;
+}
+
+// Maps both the request and the response to connect your model with API you are working with.
+const searchRequestSchema: Schema<SearchRequest, ApiRequestModel> = {
```

Comment:
I miss some talk about the features that schema has, like using paths, functions, mapping arrays, type safety...